src/vs/platform/telemetry/common/fileTelemetryService.ts

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ITelemetryService, ITelemetryInfo, ITelemetryData } from 'vs/platform/telemetry/common/telemetry';
+import * as fs from 'fs';
+import { O_CREAT, O_WRONLY } from 'constants';
+import { Disposable } from 'vs/base/common/lifecycle';
+
+/**
+ * Write telemetry into a file for test purposes
+ */
+export class FileTelemetryService extends Disposable implements ITelemetryService {
+	_serviceBrand: undefined;
+	private _isFirst = true;
+	private _enabled: boolean;
+	private _fd: number;
```

Comment:
nit: since we never actually do anything with this value I'd just have setEnabled be a no-op and get rid of this